miner/worker.go

```diff
@@ -92,7 +92,7 @@ type Result struct {
 type worker struct {
 	mu sync.Mutex
 
-	agents []Agent
+	agents map[Agent]struct{}
```

Comment:
If this is not a map, deletion looks like this:

```go
func deleteAgent(a Agent, s []Agent) []Agent {
	for i := range s {
		if s[i] == a {
			return append(s[:i], s[i+1:]...)
		}
	}
	return s
}
```

I don't want to include that. `delete` is just easier.